File: Python/timus/2008artamon.py
#!/usr/bin/env python3
def pryg(c1,c2):
  x=c2[0]-c1[0]
  y=c2[1]-c1[1]
  D=100*v**4-20*g10*y*v**2-g10**2*x**2
  return D>0
g10=98
v=int(input())
c=[]
for i in range(3):
  c.append([int(it) for it in input().split(" ")])
if pryg(c[1],c[2]) and pryg(c[2],c[1]) and (pryg(c[1],c[0]) or pryg(c[2],c[0])):
  print("Yes")
else:
  print("No")

File: C++/timus/2008artamon.cpp
#include <stdio.h>
#define g10 98
int v;
long long v2,v4;
typedef struct {
  int x,y;
} coords;
long long sqr(long long arg) {
  return arg*arg;
}
bool pryg(coords c1, coords c2) {
  int x=c2.x-c1.x,y=c2.y-c1.y;
  long long D=100*v4-20*g10*y*v2-sqr(g10)*sqr(x);
  //printf("%Ld\n",D);
  if (D>0)
    return true;
  return false;
}
int main() {
  scanf("%d",&v);
  v2=sqr(v);
  v4=sqr(v2);
  coords c[3];
  short i;
  for (i=0;i<3;i++)
    scanf("%d%d",&c[i].x,&c[i].y);
  if (pryg(c[1],c[2])&&pryg(c[2],c[1])&&(pryg(c[1],c[0])||pryg(c[2],c[0])))
    printf("Yes\n");
  else
    printf("No\n");
  return 0;
}